deployment/tfhub/object-detection/base64/object-detection-base64-predict.ipynb

```diff
@@ -1,241 +1,293 @@
+      "source": [
+        "os.environ['VERTA_HOST'] = ''\n",
+        "os.environ['VERTA_EMAIL'] = ''\n",
+        "os.environ['VERTA_DEV_KEY'] = ''"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "4eRAOufkKRw7",
+        "outputId": "64a6a46c-6827-4cbf-e96f-d279654276ee"
+      },
+      "outputs": [],
+      "source": [
+        "client = Client(os.environ['VERTA_HOST'] , debug=True)"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "EkNeEp3xKR8p",
+        "outputId": "42d72205-c7b6-4300-9c53-be39282a2759"
+      },
+      "outputs": [],
+      "source": [
+        "endpoint = client.get_or_create_endpoint('object-detection-base64')"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "XVXBGHy4KSFp"
+      },
+      "outputs": [],
+      "source": [
+        "model = endpoint.get_deployed_model()"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "PDjbzDv3KudR"
+      },
+      "source": [
+        "## Show Process Metrics"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "O0g5m4R0AOy7"
+      },
+      "outputs": [],
+      "source": [
+        "def show_metrics(images, start_time, end_time):\n",
+        "    total_time = end_time - start_time\n",
+        "    total_time = time.strftime('%Hh %Mm %Ss', time.gmtime(total_time))\n",
+        "    \n",
+        "    print(f\"Processing Time: {total_time} for {images} images.\")"
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "jgJjn6HmLLAI"
+      },
+      "source": [
+        "## Predict"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {},
+      "outputs": [],
+      "source": [
+        "def process_image(image):\n",
+        "    with open(f\"images/{image}\", 'rb') as img:\n",
+        "        img_bytes = base64.b64encode(img.read())\n",
+        "        img_str = img_bytes.decode('utf-8')\n",
+        "        img_str = json.dumps(img_str)\n",
+        "    \n",
+        "    return model.predict([image, img_str])"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "PWjaM6bY_h-R"
+      },
+      "outputs": [],
+      "source": [
+        "# n_imgs = 100\n",
+        "# n_imgs = 1000\n",
+        "n_imgs = 10000"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "wXkyYWjbLT_-"
+      },
+      "outputs": [],
+      "source": [
+        "images = os.listdir('images')"
```

Comment:
This will not work in Colab I think, because `images/` from the `deploy` notebook is not available to this `predict` notebook. Do you know if that is okay? 🤔

Reply:
You're right. It was done to run the 10k images test case (that was breaking Colab).
I refactored to download the images with `wget` and tested with a few images, it's running fine - important to remember that Colab won't be able to process a high number of images.